ambari-server/src/main/java/org/apache/ambari/server/controller/internal/BaseClusterRequest.java

```diff
@@ -224,4 +210,73 @@ public ProvisionAction getProvisionAction() {
   public void setProvisionAction(ProvisionAction provisionAction) {
     this.provisionAction = provisionAction;
   }
+
```

Comment:
This is moved here from PersistedStateImpl to take advantage of polymorphism instead of doing instanceof checks.